File: Old Solutions/April/20-04-2021/03/Invoice.java
public class Invoice {
	private String partNumber;
	private String partDescription;
	private int quantity;
	private double pricePerItem;

	public Invoice(String partNumber, String partDesciption, int quantity, double pricePerItem) {
		if (quantity < 0) {
			quantity = 0;
		}

		if (pricePerItem < 0) {
			pricePerItem = 0;
		}

		this.partNumber = partNumber;
		this.partDescription = partDesciption;
		this.quantity = quantity;
		this.pricePerItem = pricePerItem;

	}

	public String getPartNumber() {
		return this.partNumber;
	}

	public void setPartNumber(String partNumber) {
		this.partNumber = partNumber;
	}

	public String getPartDescription() {
		return this.partDescription;
	}

	public void setPartDescription(String partDescription) {
		this.partDescription = partDescription;
	}

	public int getQuantity() {
		return this.quantity;
	}

	public void setQuantity(int quantity) {
		if (quantity < 0) {
			quantity = 0;
		}
		this.quantity = quantity;
	}

	public double getPricePerItem() {
		return this.pricePerItem;
	}

	public void setPricePerItem(double pricePerItem) {
		if (pricePerItem < 0) {
			pricePerItem = 0;
		}
		this.pricePerItem = pricePerItem;
	}

	public double getInvoiceAmount() {
		return pricePerItem * quantity;
	}
}

File: Old Solutions/April/20-04-2021/03/Invoice.cpp
#include <iostream>
using namespace std;

class Invoice {
	private:
		string partNumber;
		string partDescription;
		int quantity;
		double pricePerItem;
	
	public:
		Invoice(string partNumber, string partDescription, int quantity, double pricePerItem) {
			if (quantity < 0) {
				quantity = 0;
			}

			if (pricePerItem < 0) {
				pricePerItem = 0;
			}

			this->partNumber = partNumber;
			this->partDescription = partDescription;
			this->quantity = quantity;
			this->pricePerItem = pricePerItem;
		}


		string getPartNumber() {
			return partNumber;
		}

		string getPartDescription() {
			return partDescription;
		}

		int getQuantity() {
			return quantity;
		}

		double getPricePerItem() {
			return pricePerItem;
		}

		void setPartNumber(string partNumber) {
			this->partNumber = partNumber;
		}

		void setPartDescription(string partDescription) {
			this->partDescription = partDescription;
		}

		void setQuantity(int quantity) {
			if (quantity < 0) {
				quantity = 0;
			}

			this->quantity = quantity;
		}

		void setPricePerItem(double pricePerItem) {
			if (pricePerItem < 0) {
				pricePerItem = 0;
			}

			this->pricePerItem = pricePerItem;
		}

		double getInvoiceAmount() {
			return pricePerItem * quantity;
		}
};


int main() {
	Invoice laptop("INV0100", "IBM Laptop", 3, 1920.5);
	Invoice mouse("INV0101", "Mouse Logitech", 10, 24.6);


	cout << "Invoice Amount For " << laptop.getPartDescription() << " is " << laptop.getInvoiceAmount() << endl;
	cout << "Invoice Amount For " << mouse.getPartDescription() << " is " << mouse.getInvoiceAmount() << endl;

	return 0;
}